fuzztest/internal/type_support.h

```diff
@@ -54,33 +54,51 @@ decltype(auto) AutodetectTypePrinter();
 template <typename T>
 constexpr bool HasKnownPrinter();
 
-// If `needle` is present in `haystack`, consume everything until `needle` and
-// return true. Otherwise, return false.
-inline bool ConsumePrefixUntil(absl::string_view& haystack,
-                               absl::string_view needle) {
-  size_t pos = haystack.find(needle);
-  if (pos == haystack.npos) return false;
-  haystack.remove_prefix(pos + needle.size());
+// If `prefix` is present in `name`, consume everything until the rightmost
+// occurrence of `prefix` and return true. Otherwise, return false.
+constexpr bool ConsumePrefixUntil(absl::string_view& name,
+                                  absl::string_view prefix) {
+  size_t pos = name.rfind(prefix);
+  if (name.npos == pos) return false;
+  name.remove_prefix(pos + prefix.size());
   return true;
 }
 
-inline void SkipAnonymous(absl::string_view& in) {
-  while (ConsumePrefixUntil(in, "(anonymous namespace)::")) {
+constexpr bool ConsumeUnnecessaryNamespacePrefix(absl::string_view& name) {
+  constexpr absl::string_view prefixes[] = {
+      // GCC adds the function name in which the type was defined e.g.,
+      // MyFunction()::MyStruct{}.
+      "()::",
+      // Various anonymous namespace prefixes different compilers use:
+      "{anonymous}::",
+      "(anonymous namespace)::",
+      "<unnamed>::",
+  };
+  for (absl::string_view p : prefixes) {
+    if (ConsumePrefixUntil(name, p)) return true;
```

Comment:
I think it's incorrect and we should iterate over all of them. Because multiple prefixes can be present at the same time

Reply:
Thanks for checking!

Yes, multiple prefixes can be present, e.g., we could have an `anonymous::bla::bla::MyFunction()::MyStruct()`, and if we removed the `anonymous::` part first and then stopped, we'd still have the `::bla::bla::MyFunction()::`. But this is why we start with matching `<whatever>()::` first: see comment in the `prefixes` list above. 

Do you think this works?